packages/utils/src/media/measureCanvasContent.ts

```diff
@@ -0,0 +1,80 @@
+import type { ICanvas } from '@pixi/settings';
+
+function checkRow(data: Uint8ClampedArray, width: number, y: number)
+{
+    for (let x = 0, index = 4 * y * width; x < width; ++x, index += 4)
+    {
+        if (data[index + 3] !== 0) return false;
+    }
+
+    return true;
+}
+
+function checkColumn(data: Uint8ClampedArray, width: number, x: number, top: number, bottom: number)
+{
+    const stride = 4 * width;
+
+    for (let y = top, index = (top * stride) + (4 * x); y <= bottom; ++y, index += stride)
+    {
+        if (data[index + 3] !== 0) return false;
+    }
+
+    return true;
+}
+
+/**
+ * Measure canvas content (size/bounds of non-transparent pixels)
+ * @memberof PIXI.utils
+ * @function measureCanvasContent
+ * @param {PIXI.ICanvas} canvas - the canvas to measure
+ * @returns {object} Canvas measurement
+ */
+export function measureCanvasContent(canvas: ICanvas):
+{
+    size: {
+        width: number;
+        height: number;
+    }
+    bounds: {
+        top: number;
+        bottom: number;
+        left: number;
+        right: number;
+    }
+}
```

Comment:
Would be nice to make a type here for the result. Also, I'd suggest flattening this, the `size` and `bounds` levels aren't really necessary and this is not an interface that's likely to expand at all so that structure just kinda gets in the way.

Having as a type means that we can document it:

```ts
/**
 * Result when using `utils.measureCanvasContent` to determine
 * the bounds of a canavas' visual content excluding transparency.
 * @memberof PIXI.utils
 */
interface CanvasContent {
    /** Total width in pixels of the canvas */
    width: number;
    /** Total height in pixels of the canvas */
    height: number;
    /** Offset from the top, in pixels, where the canvas' content begins */
    top: number;
    /** Offset from the bottom, in pixels, where the canvas' content begins */
    bottom: number;
    /** Offset from the left, in pixels, where the canvas' content begins */
    left: number;
    /** Offset from the right, in pixels, where the canvas' content begins */
    right: number;
}
```

Reply:
Updated!

Reply:
I am not very satisfied with the wording `Offset from the bottom, in pixels, where the canvas' content begins`, it is more like `height - bottom`.

Also the name `CanvasContent` is not very good in my opinion, since it only contains the pixel boundary information of the canvas, not the real "content".

Or maybe we can use the existing class `Rectangle` in `@pixi/math`?

Reply:
feel free to rename/refactor to whatever you see fit.

Reply:
@SuperSodaSea I'm cool if you want to take a pass on the naming/types. I'm not super opinionated about it. Just having a type that's flatter and documented is my preference.

Reply:
perhaps call it `getCanvasBoundingBox()` returning a `BoundingBox`.

related:
* [Minimum bounding rectangle](https://en.wikipedia.org/wiki/Minimum_bounding_rectangle)
* [Element.getBoundingClientRect()](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect)